File: packages/dioxus-ui/src/policy_display.rs
fn normalize(raw: Option<&str>) -> String {
    raw.unwrap_or("").trim().to_lowercase()
}

pub fn display_stealth_mode(raw: Option<&str>) -> &'static str {
    let v = normalize(raw);
    match v.as_str() {
        "" => "Auto",
        "off" => "Off",
        "manual" => "Manual",
        "performance" | "base" => "Performance",
        "stealth" => "Stealth",
        "anti-dpi" | "antidpi" | "max" | "stealthmax" | "stealth-max" => "AntiDPI",
        "auto" | "intelligent" => "Auto",
        _ => "Auto",
    }
}

pub fn display_fec_mode(raw: Option<&str>) -> &'static str {
    let v = normalize(raw);
    match v.as_str() {
        "off" | "zero" => "Off",
        _ => "Auto",
    }
}

pub fn display_cc_mode(raw: Option<&str>) -> &'static str {
    let v = normalize(raw);
    match v.as_str() {
        "" | "server" => "BBR3",
        "reno" => "RENO",
        "bbr2" => "BBR2",
        "bbr3" => "BBR3",
        _ => "Custom",
    }
}

pub fn display_mtu(raw: Option<&str>) -> String {
    let v = raw.unwrap_or("").trim();
    if v.is_empty() || v.to_lowercase() == "server" {
        return "1200".into();
    }
    if v.chars().all(|c| c.is_ascii_digit()) {
        v.into()
    } else {
        "1200".into()
    }
}

File: apps/svelte-desktop/src/lib/policy-display.ts
import { congestionControlDisplayLabel } from "@quicfuscate/ui/congestion-control";

function normalize(raw: string | null | undefined): string {
  return (raw ?? "").trim().toLowerCase();
}

export function displayStealthMode(raw: string | null | undefined): string {
  const v = (raw ?? "").trim();
  if (
    v === "off" ||
    v === "manual" ||
    v === "performance" ||
    v === "stealth" ||
    v === "Stealth MAX" ||
    v === "dynamic"
  ) {
    return v;
  }
  return v || "dynamic";
}

export function displayFecMode(raw: string | null | undefined): string {
  const v = normalize(raw);
  if (v === "off" || v === "zero") return "Off";
  return "Auto";
}

export function displayCcMode(raw: string | null | undefined): string {
  const v = normalize(raw);
  if (!v || v === "server") return "BBR3";
  return congestionControlDisplayLabel(v);
}

export function displayMtu(raw: string | null | undefined): string {
  const v = (raw ?? "").trim();
  if (!v || v.toLowerCase() === "server") return "1200";
  if (/^\d+$/.test(v)) return v;
  return "1200";
}
